js/webtorrent/components/mediaViewer.js

```diff
@@ -19,24 +19,26 @@ const SUPPORTED_AUDIO_EXTENSIONS = [
 
 module.exports = class MediaViewer extends React.Component {
   render () {
-    const torrent = this.props.torrent
-    const ix = this.props.ix
+    const { torrent, serverUrl, ix } = this.props
+
     const file = torrent.files[ix]
+    const fileURL = serverUrl && (serverUrl + '/' + ix)
+
     const fileExt = file && getExtension(file.name)
     const isVideo = SUPPORTED_VIDEO_EXTENSIONS.includes(fileExt)
     const isAudio = SUPPORTED_AUDIO_EXTENSIONS.includes(fileExt)
-    const fileURL = torrent.serverURL && (torrent.serverURL + '/' + ix)
 
     let content
-    if (torrent.serverURL == null) {
+    if (!file || !serverUrl) {
```

Comment:
Fix race condition where media is briefly added to <iframe>. If there is no file object yet, but there is a serverUrl, then an iframe is added to the page, instead of a "Loading..." message.